rs/rosetta-api/icrc1/tests/golden_state_upgrade_downgrade.rs

```diff
@@ -0,0 +1,146 @@
+const GOLDDAO: (&str, &str) = ("tyyy3-4aaaa-aaaaq-aab7a-cai", "GoldDAO");
+const HOTORNOT: (&str, &str) = ("6rdgd-kyaaa-aaaaq-aaavq-cai", "HotOrNot");
+const ICGHOST: (&str, &str) = ("4c4fd-caaaa-aaaaq-aaa3a-cai", "ICGhost");
+const ICLIGHTHOUSE: (&str, &str) = ("hhaaz-2aaaa-aaaaq-aacla-cai", "ICLighthouse");
+const ICPANDA: (&str, &str) = ("druyg-tyaaa-aaaaq-aactq-cai", "ICPanda");
+const ICPCC: (&str, &str) = ("lrtnw-paaaa-aaaaq-aadfa-cai", "ICPCC");
+const ICX: (&str, &str) = ("rffwt-piaaa-aaaaq-aabqq-cai", "ICX");
+const KINIC: (&str, &str) = ("73mez-iiaaa-aaaaq-aaasq-cai", "Kinic");
+const DECIDEAI: (&str, &str) = ("xsi2v-cyaaa-aaaaq-aabfq-cai", "DecideAI");
+const MOTOKO: (&str, &str) = ("k45jy-aiaaa-aaaaq-aadcq-cai", "Motoko");
+const NEUTRINITE: (&str, &str) = ("f54if-eqaaa-aaaaq-aacea-cai", "Neutrinite");
+const NUANCE: (&str, &str) = ("rxdbk-dyaaa-aaaaq-aabtq-cai", "Nuance");
+const OPENFPL: (&str, &str) = ("ddsp7-7iaaa-aaaaq-aacqq-cai", "OpenFPL");
+const ORIGYN: (&str, &str) = ("lkwrt-vyaaa-aaaaq-aadhq-cai", "Origyn");
+const SNEED: (&str, &str) = ("hvgxa-wqaaa-aaaaq-aacia-cai", "Sneed");
+const SONIC: (&str, &str) = ("qbizb-wiaaa-aaaaq-aabwq-cai", "Sonic");
+const TRAX: (&str, &str) = ("emww2-4yaaa-aaaaq-aacbq-cai", "Trax");
+const WATERNEURON: (&str, &str) = ("jcmow-hyaaa-aaaaq-aadlq-cai", "WaterNeuron");
+const YUKU: (&str, &str) = ("atbfz-diaaa-aaaaq-aacyq-cai", "Yuku");
+
+#[test]
+fn should_upgrade_icrc_ck_canisters_with_golden_state() {
+    let ledger_wasm = ledger_wasm();
+    let ledger_wasm_u256 = ledger_wasm_u256();
+
+    let canisters = vec![CK_BTC_LEDGER];
+    let canisters_u256 = vec![
+        CK_ETH_LEDGER,
+        CK_USDC_LEDGER,
+        CK_LINK_LEDGER,
+        CK_OCT_LEDGER,
+        CK_PEPE_LEDGER,
+        CK_SHIB_LEDGER,
+    ];
+
+    let state_machine = new_state_machine_with_golden_fiduciary_state_or_panic();
+
+    for canister in canisters {
+        upgrade_canister(&state_machine, canister, ledger_wasm.clone());
+    }
+    for canister_u256 in canisters_u256 {
+        upgrade_canister(&state_machine, canister_u256, ledger_wasm_u256.clone());
+    }
+}
+
+#[test]
+fn should_upgrade_icrc_sns_canisters_with_golden_state() {
+    let ledger_wasm = ledger_wasm();
+
+    let canisters = vec![
+        DRAGGINZ,
+        OPENCHAT,
+        ICPSWAP,
+        BOOMDAO,
+        CATALYZE,
+        ELNAAI,
+        ESTATEDAO,
+        GOLDDAO,
+        HOTORNOT,
+        ICGHOST,
+        ICLIGHTHOUSE,
+        ICPANDA,
+        ICPCC,
+        ICX,
+        KINIC,
+        DECIDEAI,
+        MOTOKO,
+        NEUTRINITE,
+        NUANCE,
+        OPENFPL,
+        ORIGYN,
+        SNEED,
+        SONIC,
+        TRAX,
+        WATERNEURON,
+        YUKU,
+    ];
+
+    let state_machine = new_state_machine_with_golden_sns_state_or_panic();
+
+    for canister in canisters {
+        upgrade_canister(&state_machine, canister, ledger_wasm.clone());
```

Comment:
nit, and no change required, but "canister" usually means "canister_id" or a struct representing a canister, not a tuple.

Reply:
Since I was anyway making changes I renamed these also, thanks for the suggestion!